internal/pkg/cli/env_upgrade.go

```diff
@@ -18,21 +32,56 @@ type envUpgradeVars struct {
 // and clients to execute the command.
 type envUpgradeOpts struct {
 	envUpgradeVars
+
+	store environmentStore
+	sel   appEnvSelector
 }
 
-func newEnvUpgradeOpts(vars envUpgradeVars) *envUpgradeOpts {
+func newEnvUpgradeOpts(vars envUpgradeVars) (*envUpgradeOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("connect to config store: %v", err)
+	}
 	return &envUpgradeOpts{
 		envUpgradeVars: vars,
-	}
+		store:          store,
+		sel:            selector.NewSelect(prompt.New(), store),
+	}, nil
 }
 
 // Validate returns an error if the values passed by flags are invalid.
 func (o *envUpgradeOpts) Validate() error {
+	if o.all && o.name != "" {
+		return fmt.Errorf("cannot specify both --%s and --%s flags", allFlag, nameFlag)
+	}
+	if o.name != "" {
+		if _, err := o.store.GetEnvironment(o.appName, o.name); err != nil {
+			var errEnvDoesNotExist *config.ErrNoSuchEnvironment
```

Comment:
Why do we want to differentiate this error? Is it for making the error message more explicit?

Reply:
yup yup so that the errs are not redundant